### DISCUSSION TITLE
Convert between durations?

### DISCUSSION THREAD

**Discussion by tisonkun** [apache/arrow-rs#6425] [Q&A] [answered]
Timestamps can be converted between different time unit, but Durations can only be reinterpreted from/to int64.

Is it possible to convert between Durations in different time unit?

```rust
pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
...
        (Timestamp(_, _), Timestamp(_, _) | ...) => true,
...
        (Int64, Duration(_)) => true,
        (Duration(_), Int64) => true,
...
}
```

**Comment by tisonkun** [comment]
cc @alamb @tustvold or if you know who are familiar with this sort of APIs.

**Reply by tustvold** [reply, marked as answer]
Adding support to the cast kernels for these makes sense to me and can likely follow the pattern for Timestamps

**Reply by alamb** [reply]
I agree converting durations makes sense. 

Here is a similar request from @simonvandel  in DataFusion: https://github.com/apache/datafusion/issues/11097

@nrc  has done some work recently to improve duration support as well here https://github.com/apache/arrow-rs/issues/6245

**Reply by tisonkun** [reply]
I'm preparing a PR now.

 UPDATE - https://github.com/apache/arrow-rs/pull/6452
